### PR TITLE
BD-66 feat: avatar component

### PR DESCRIPTION
**Pull request by sfgardev**

### CODE REVIEW

File: pages/profile/[id].tsx [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
+    <section className='py-9 pl-6 pr-16'>
+      <div className='flex items-start justify-between'>
+        <Avatar
+          alt={publicProfileData?.userName}
+          imageUrl={publicProfileData?.avatars[0]?.url}
+          isLoading={isLoadingPublicProfile}
+        />
+        {isCurrentUser && (
+          <Button asChild variant='secondary'>
+            <Link href='/profile/settings'>{t.profile.button.profileSettings}</Link>
```
Comment (SlavaMryasov) [inline pages/profile/[id].tsx:35]: use profileId, your path is hardcode

File: pages/profile/[id].tsx [hunk 2]
```diff
@@ -1,24 +1,42 @@
...
+  const router = useRouter()
+  const profileId = Number(router.query.id)
+
+  const { data: publicProfileData, isLoading: isLoadingPublicProfile } =
+    publicUserApi.useGetPublicProfileByIdQuery({ profileId })
+
+  const isCurrentUser = authMeData?.userId === publicProfileData?.id
+
+  if (!profileId) {
+    return <p className='text-light-100'>Profile id is invalid</p>
```
Comment (SlavaMryasov) [inline pages/profile/[id].tsx:22]: u should use value from internationalization but no hardcode

File: services/public-user/publicUserSlice.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { baseQueryWithReauth } from '@/services/baseQuery'
+import { PublicUserProfileResponse } from '@/services/public-user/types'
+import { createApi } from '@reduxjs/toolkit/query/react'
+
+export const publicUserApi = createApi({
+  baseQuery: baseQueryWithReauth,
+  endpoints: builder => ({
+    getPublicProfileById: builder.query<PublicUserProfileResponse, { profileId: number }>({
+      query: ({ profileId }) => ({
+        url: `public-user/profile/${profileId}`,
```
Comment (SlavaMryasov) [inline services/public-user/publicUserSlice.ts:10]: if we wrote "method" in other slices, perhaps we need to write it here too?

File: services/public-user/types.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export type PublicUserProfileResponse = {
+  aboutMe: string
+  avatars: Avatar[]
+  id: number
+  userName: string
+}
+
+export type Avatar = {
```
Comment (SlavaMryasov) [inline services/public-user/types.ts:8]: this type is already in 'services/profile'

**Review by SlavaMryasov — CHANGES_REQUESTED** [review]

File: pages/profile/[id].tsx (hunk 1 above)
Comment (sfgardev) [reply pages/profile/[id].tsx:35]: for what purpose shouild i use profileId?

File: services/public-user/publicUserSlice.ts (hunk 3 above)
Comment (sfgardev) [reply services/public-user/publicUserSlice.ts:10]: it is like placeholder here, in future we should redirect to 404 page, so no need to translate it

File: pages/profile/[id].tsx (hunk 2 above)
Comment (sfgardev) [reply pages/profile/[id].tsx:22]: it is like placeholder here, in future we should redirect to 404 page, so no need to translate it

File: services/public-user/publicUserSlice.ts (hunk 3 above)
Comment (sfgardev) [reply services/public-user/publicUserSlice.ts:10]: we need to use method only for mutation, no need to write method in query

File: services/public-user/types.ts (hunk 4 above)
Comment (sfgardev) [reply services/public-user/types.ts]: done

**Review by SlavaMryasov — APPROVED** [review]

**Pull request merged**